src/snapshot.rs

```diff
@@ -52,6 +57,26 @@ pub struct StartupData {
   raw_size: int,
 }
 
+impl From<Box<[u8]>> for StartupData {
+  fn from(slice: Box<[u8]>) -> Self {
+    let mut data: Vec<u8> = vec![];
+    data.copy_from_slice(&slice);
+    StartupData {
+      data: data.as_ptr() as *const char,
+      raw_size: data.len() as int,
+    }
+  }
+}
+
+impl From<&[u8]> for StartupData {
+  fn from(slice: &[u8]) -> Self {
+    StartupData {
+      data: slice.as_ptr() as *const char,
+      raw_size: slice.len() as int,
+    }
+  }
+}
```

Comment:
AFAIK You can create a `StartupData` only through `SnapshotCreator::CreateBlob` and V8 apis. See https://chromium.googlesource.com/v8/v8/+/refs/heads/main/src/snapshot/snapshot.cc#740